### PR TITLE
[curl] Fix brototli option flag

### PR DESCRIPTION
**Pull request by aahmed-se**

### CODE REVIEW

File: var/spack/repos/builtin/packages/curl/package.py [hunk 1]
```diff
@@ -72,16 +75,20 @@ def configure_args(self):
...
             '--without-libgsasl',
             '--without-libmetalink',
             '--without-libpsl',
             '--without-zstd',
         ]
 
+        if spec.satisfies('+libidn2'):
+            args.append('--with-libidn2=' + spec['libidn2'].prefix)
+        else:
+            args.append('--without-libidn2')
```
Comment (tgamblin) [inline var/spack/repos/builtin/packages/curl/package.py]: can you replace this with:

```python
args += self.with_or_without(libidn2, 'prefix')
```

See https://spack.readthedocs.io/en/latest/build_systems/autotoolspackage.html#helper-functions

File: var/spack/repos/builtin/packages/curl/package.py [hunk 2]
```diff
@@ -92,6 +99,12 @@ def configure_args(self):
         else:
             args.append('--without-gssapi')
 
+        if spec.satisfies('~librtmp'):
+            args.append('--without-librtmp')
```
Comment (tgamblin) [inline var/spack/repos/builtin/packages/curl/package.py]: Same here but no argument:

```python
args += self.with_or_without("librtmp")
```

File: var/spack/repos/builtin/packages/curl/package.py [hunk 3]
```diff
@@ -92,6 +99,12 @@ def configure_args(self):
...
+
+        if spec.satisfies('~ldap'):
+            args.append('--disable-ldap')
```
Comment (tgamblin) [inline var/spack/repos/builtin/packages/curl/package.py]: Similar here (see link above):

```python
args += self.enable_or_disable("ldap") 
```

**Review by tgamblin — CHANGES_REQUESTED** [review]
LGTM -- just a request to use helper functions where possible.

File: var/spack/repos/builtin/packages/curl/package.py (hunk 3 above)
Comment (aahmed-se) [reply var/spack/repos/builtin/packages/curl/package.py]: done

File: var/spack/repos/builtin/packages/curl/package.py (hunk 2 above)
Comment (aahmed-se) [reply var/spack/repos/builtin/packages/curl/package.py]: done

File: var/spack/repos/builtin/packages/curl/package.py (hunk 1 above)
Comment (aahmed-se) [reply var/spack/repos/builtin/packages/curl/package.py]: done

File: var/spack/repos/builtin/packages/curl/package.py [hunk 4]
```diff
@@ -48,6 +48,9 @@ class Curl(AutotoolsPackage):
     variant('libssh',     default=False, description='enable libssh support')  # , when='7.58:')
     variant('darwinssl',  default=sys.platform == 'darwin', description="use Apple's SSL/TLS implementation")
     variant('gssapi',     default=False, description='enable Kerberos support')
+    variant('librtmp',    default=False, description='enable Rtmp support')
+    variant('ldap',       default=False, description='enable ldap support')
+    variant('libidn2',    default=True,  description='enable libidn2 support')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/curl/package.py]: I know this was previously a required dependency, but should we default this to False to minimize the number of deps?

Comment (aahmed-se) [reply var/spack/repos/builtin/packages/curl/package.py]: I don't mind either way, what's the guideline . To minimize dependencies in default settings ?

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/curl/package.py]: There isn't a strict guideline at the moment. Right now, Spack builds all packages from source code and there are no public binaries for packages, so I tend to disable most variants, especially the ones that add additional dependencies. Once public binaries become available, this guidance may change and we'll likely default to the most commonly used/useful variants being enabled by default.

Comment (aahmed-se) [reply var/spack/repos/builtin/packages/curl/package.py]: changed to false

**Review by adamjstewart — APPROVED** [review]

**Review by tgamblin — APPROVED** [review]

**Pull request merged**